src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -332,14 +333,18 @@ export class NodeClientRequest extends ClientRequest {
     )
     this.log('written request body:', writtenRequestBody)
 
-    const finalRequestBody = bodyBufferToString(
-      Buffer.concat(
-        chunk ? concatChunkToBuffer(chunk, this.requestBody) : this.requestBody
-      )
+    // Write the last request body chunk to the internal request body buffer.
+    if (chunk) {
+      this.requestBody = concatChunkToBuffer(chunk, this.requestBody)
```

Comment:
Concatenating the last chunk provided to `.end()` so it becomes a part of the `this.requestBody`.